### PR TITLE
fix(surveys): enable button text per question instead

### PR DESCRIPTION
**Pull request by liyiy**
## Problem

We only had 1 field to set up a "submit button text" value, which doesn't work for multiple question surveys.

## Changes
- All new surveys will be using the "button text" field per question instead
- Continue to support previous button text for existing surveys that don't have new field

https://github.com/PostHog/posthog-js/pull/893

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Comment by liyiy** [comment]
JS code fix to follow, this should be review-able right now though!

**Review by neilkakkar — APPROVED** [review]
Makes sense, looks good 👍

Not sure if just missing it, or will come in a follow up, but I don't yet see how someone can add the button text in the UI?

File: frontend/src/scenes/surveys/SurveyEdit.tsx [hunk 1]
```diff
@@ -518,6 +525,14 @@ export default function SurveyEdit(): JSX.Element {
                                                                         </Field>
                                                                     </div>
                                                                 )}
+                                                                <Field name="buttonText" label="Button text">
+                                                                    <LemonInput
+                                                                        value={
+                                                                            question.buttonText ||
```
Comment (neilkakkar) [inline frontend/src/scenes/surveys/SurveyEdit.tsx:531]: oh nvm found it!

The `||` here means whenever I empty the buttonText it is immediately replaced by the text in `submitButtonText` - probably annoying for existing users, is this for backwards compatibility?

Comment (liyiy) [reply frontend/src/scenes/surveys/SurveyEdit.tsx:531]: yes for backwards compatibility, let me see if I can fix that though

**Pull request merged**